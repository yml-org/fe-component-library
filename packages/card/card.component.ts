import { CardComponent } from './src/card/card.component';

declare global {
  interface HTMLElementTagNameMap {
    'ymlwebcl-card': CardComponent;
  }
}
