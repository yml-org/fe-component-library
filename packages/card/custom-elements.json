{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "card.component.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/index.ts",
      "declarations": [],
      "exports": [
        {
          "kind": "js",
          "name": "*",
          "declaration": {
            "name": "*",
            "package": "./card/card.component"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "dist/assets/index.js",
      "declarations": [
        {
          "kind": "variable",
          "name": "e"
        },
        {
          "kind": "variable",
          "name": "i"
        },
        {
          "kind": "variable",
          "name": "n"
        },
        {
          "kind": "variable",
          "name": "f",
          "default": "class extends Ht(null) {\n  constructor() {\n    super(...arguments), this.hasRoundedBorder = !1, this.customSlotName = \"\", this.showCustomSlot = !1, this.cardHeight = 300, this.cardWidth = 300, this.cardDimensionUnit = dt.Pixel, this.isHorizontal = !1;\n  }\n  getContainerClassList() {\n    const { hasRoundedBorder: o, isHorizontal: t } = this;\n    return `webcl-card-wrapper flex ${t ? \"flex-row\" : \"flex-col\"} ${o ? \"rounded-lg\" : \"\"}`.trim();\n  }\n  setCardDimensions() {\n    const { cardHeight: o, cardWidth: t, cardDimensionUnit: e } = this;\n    return `width:${t}${e}; min-height:${o}${e}`;\n  }\n  renderSlots() {\n    const { showCustomSlot: o } = this;\n    return o ? I`<slot name=${this.customSlotName} />` : I`<div class=\"webcl-card-header-section\" part=\"webcl-card-header\">\n            <slot name=\"webcl-card-header\" />\n          </div>\n          <div class=\"webcl-card-content-section\" part=\"webcl-card-content\">\n            <slot name=\"webcl-card-content\" />\n          </div>\n          <div\n            class=\"webcl-card-footer-section mt-auto\"\n            part=\"webcl-card-footer\"\n          >\n            <slot name=\"webcl-card-footer\" />\n          </div>`;\n  }\n  render() {\n    return I`<div\n      class=${this.getContainerClassList()}\n      style=${this.setCardDimensions()}\n      part=\"webcl-card-container\"\n    >\n      ${this.renderSlots()}\n    </div>`;\n  }\n}"
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "i",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "n",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "js",
          "name": "CardComponent",
          "declaration": {
            "name": "f",
            "module": "dist/assets/index.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/card/card.component.spec.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/card/card.component.ts",
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "CardComponent",
          "members": [
            {
              "kind": "field",
              "name": "hasRoundedBorder",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false"
            },
            {
              "kind": "field",
              "name": "customSlotName",
              "type": {
                "text": "string | undefined"
              },
              "default": "''"
            },
            {
              "kind": "field",
              "name": "showCustomSlot",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false"
            },
            {
              "kind": "field",
              "name": "cardHeight",
              "type": {
                "text": "number | undefined"
              },
              "default": "300"
            },
            {
              "kind": "field",
              "name": "cardWidth",
              "type": {
                "text": "number | undefined"
              },
              "default": "300"
            },
            {
              "kind": "field",
              "name": "cardDimensionUnit",
              "type": {
                "text": "CardDimensionUnit | undefined"
              }
            },
            {
              "kind": "field",
              "name": "isHorizontal",
              "type": {
                "text": "boolean | undefined"
              },
              "default": "false"
            },
            {
              "kind": "method",
              "name": "getContainerClassList",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "setCardDimensions",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "renderSlots",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "render",
              "privacy": "public"
            },
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]",
              "inheritedFrom": {
                "name": "TailwindElement",
                "module": "src/shared/tailwind.element.ts"
              }
            }
          ],
          "mixins": [
            {
              "name": "TailwindElement",
              "module": "/src/shared/tailwind.element"
            }
          ],
          "superclass": {
            "module": "src/card/card.component.ts"
          }
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "CardComponent",
          "declaration": {
            "name": "CardComponent",
            "module": "src/card/card.component.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/constants/card.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/constants/card.variants.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "cardWithRoundedBorder",
          "type": {
            "text": "CardVariant"
          },
          "default": "{\n  hasRoundedBorder: true,\n}"
        },
        {
          "kind": "variable",
          "name": "cardWithCustomWidthHeight",
          "type": {
            "text": "CardVariant"
          },
          "default": "{\n  ...cardWithRoundedBorder,\n  cardHeight: 450,\n  cardWidth: 300,\n}"
        },
        {
          "kind": "variable",
          "name": "cardWithCustomSlot",
          "type": {
            "text": "CardVariant"
          },
          "default": "{\n  ...cardWithCustomWidthHeight,\n  customSlotName: 'custom-card-slot',\n  showCustomSlot: true,\n}"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "cardWithRoundedBorder",
          "declaration": {
            "name": "cardWithRoundedBorder",
            "module": "src/constants/card.variants.ts"
          }
        },
        {
          "kind": "js",
          "name": "cardWithCustomWidthHeight",
          "declaration": {
            "name": "cardWithCustomWidthHeight",
            "module": "src/constants/card.variants.ts"
          }
        },
        {
          "kind": "js",
          "name": "cardWithCustomSlot",
          "declaration": {
            "name": "cardWithCustomSlot",
            "module": "src/constants/card.variants.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/shared/tailwind.element.ts",
      "declarations": [
        {
          "kind": "mixin",
          "description": "",
          "name": "TailwindElement",
          "members": [
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]"
            }
          ],
          "parameters": [
            {
              "name": "style",
              "type": {
                "text": "CSSStyleSheet | null"
              }
            }
          ]
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "TailwindElement",
          "declaration": {
            "name": "TailwindElement",
            "module": "src/shared/tailwind.element.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/types/card.ts",
      "declarations": [],
      "exports": []
    }
  ]
}
