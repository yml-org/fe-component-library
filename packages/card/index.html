<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>web components: Card</title>
  </head>
  <body>
    <ymlwebcl-card></ymlwebcl-card>

    <script type="module" src="src/card/card.component.ts"></script>
  </body>
</html>
