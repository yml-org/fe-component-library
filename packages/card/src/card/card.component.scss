.storybook-card-container {
  card-component {
    &::part(webcl-card-container) {
      border: 1px solid black;
    }
  }

  &.custom-header-variant {
    card-component {
      &::part(webcl-card-header) {
        font-size: 1.5rem;
      }
    }
  }

  &.all-slots-variant {
    card-component {
      &::part(webcl-card-container) {
        padding: 1rem;
      }

      &::part(webcl-card-header) {
        font-size: 1.5rem;
        margin-bottom: 1.5rem;
      }

      p {
        margin-top: 1.25rem;
      }
    }
  }
}