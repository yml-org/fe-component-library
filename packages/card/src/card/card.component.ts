import { html, nothing } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { TailwindElement } from '../shared/tailwind.element';
import { msg, str } from '@lit/localize';

@customElement('card-component')
export class CardComponent extends TailwindElement(null) {
  render() {
    return html`<div
      class="webcl-card-wrapper"
      part="webcl-card-container"
    ></div>`;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'card-component': CardComponent;
  }
}
