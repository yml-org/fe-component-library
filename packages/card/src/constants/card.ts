export enum CardDimensionUnits {
  Pixel = 'px',
  Centimeter = 'cm',
  Millimeter = 'mm',
  Inch = 'in',
  Points = 'pt',
  Picas = 'pc',
  Em = 'em',
  Rem = 'rem',
  Ex = 'ex',
  Ch = 'ch',
  Vw = 'vw',
  Vh = 'vh',
  Percentage = '%',
}
