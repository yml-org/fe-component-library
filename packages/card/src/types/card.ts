export type CardDimensionUnit =
  | 'px'
  | '%'
  | 'em'
  | 'rem'
  | 'pt'
  | 'cm'
  | 'in'
  | 'mm'
  | 'pc'
  | 'ex'
  | 'ch'
  | 'vh'
  | 'vw';

export type CardVariant = {
  hasRoundedBorder?: boolean;
  customSlotName?: string;
  showCustomSlot?: boolean;
  cardHeight?: number;
  cardWidth?: number;
  cardDimensionUnit?: CardDimensionUnit;
  isHorizontal?: boolean;
};
