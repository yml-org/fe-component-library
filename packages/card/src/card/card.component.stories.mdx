import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './card.component';
import {
  cardWithRoundedBorder,
  cardWithCustomWidthHeight,
  cardWithCustomSlot,
} from '../constants/card.variants.ts';
import './card.component.scss';

<Meta title="Components/Card" />

# Card Component

The card component is a Lit component that renders a skeleton card that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name         | Type              | Description                                                |
| ----------------- | ----------------- | ---------------------------------------------------------- |
| hasRoundedBorder  | boolean           | Flag to determine whether or not to have rounded borders   |
| cardWidth         | number            | The width of the card component                            |
| cardHeight        | number            | The height of the card component                           |
| customSlotName    | string            | The name of the child slot                                 |
| showCustomSlot    | boolean           | Flag to determine whether or not to render the custom slot |
| cardDimensionUnit | CardDimensionUnit | The unit for the card width and height                     |
| isHorizontal      | boolean           | Flag to determine the content direction                    |

Following are the values for the CardDimensionUnit:

1. px
2. %
3. em
4. rem
5. pt
6. cm
7. in
8. mm
9. pc
10. ex
11. ch
12. vh
13. vw

The default slots provided in the component are as follows:

1. webcl-card-header
2. webcl-card-content
3. webcl-card-footer

# Examples

Basic card component without `props`

export const TemplateDefault = () =>
  html`<div class="storybook-card-container">
    <ymlwebcl-card></ymlwebcl-card>
  </div>`;

<Canvas columns={1}>
  <Story name="default">{TemplateDefault.bind({})}</Story>
</Canvas>

Basic card with rounded border

export const withRoundedBorderTemplate = ({ hasRoundedBorder }) =>
  html`<div class="storybook-card-container">
    <ymlwebcl-card .hasRoundedBorder=${hasRoundedBorder}></ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="with rounded border" args={cardWithRoundedBorder}>
    {withRoundedBorderTemplate.bind({})}
  </Story>
</Canvas>

Basic card with custom height and width

export const withCustomHeightWidth = ({
  hasRoundedBorder,
  cardWidth,
  cardHeight,
}) =>
  html`<div class="storybook-card-container">
    <ymlwebcl-card
      .hasRoundedBorder=${hasRoundedBorder}
      .cardHeight=${cardHeight}
      .cardWidth=${cardWidth}
    ></ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="with custom height and width" args={cardWithCustomWidthHeight}>
    {withCustomHeightWidth.bind({})}
  </Story>
</Canvas>

Basic card using defined header slot

export const withDefinedHeaderSlot = ({
  hasRoundedBorder,
  cardWidth,
  cardHeight,
}) =>
  html`<div class="storybook-card-container custom-header-variant">
    <ymlwebcl-card
      .hasRoundedBorder=${hasRoundedBorder}
      .cardHeight=${cardHeight}
      .cardWidth=${cardWidth}
    >
      <div slot="webcl-card-header">
        <h1>WebCl Card Header</h1>
      </div>
    </ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="using defined header slot" args={cardWithCustomWidthHeight}>
    {withDefinedHeaderSlot.bind({})}
  </Story>
</Canvas>

Basic card using defined footer slot

export const withDefinedFooterSlot = ({
  hasRoundedBorder,
  cardWidth,
  cardHeight,
}) =>
  html`<div class="storybook-card-container">
    <ymlwebcl-card
      .hasRoundedBorder=${hasRoundedBorder}
      .cardHeight=${cardHeight}
      .cardWidth=${cardWidth}
    >
      <div slot="webcl-card-footer">
        <p>WebCl Card Footer</p>
      </div>
    </ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="using defined footer slot" args={cardWithCustomWidthHeight}>
    {withDefinedFooterSlot.bind({})}
  </Story>
</Canvas>

Basic card using all the defined slots

export const withAllDefinedSlots = ({
  hasRoundedBorder,
  cardWidth,
  cardHeight,
}) =>
  html`<div class="storybook-card-container all-slots-variant">
    <ymlwebcl-card
      .hasRoundedBorder=${hasRoundedBorder}
      .cardHeight=${cardHeight}
      .cardWidth=${cardWidth}
    >
      <div slot="webcl-card-header">
        <h1>WebCl Card Header</h1>
      </div>
      <div slot="webcl-card-content">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
          ac aliquam orci. In dignissim eget enim ut faucibus. Donec volutpat
          libero a ipsum tincidunt, in vehicula tellus congue. Mauris id
          consequat justo. Donec turpis justo, dictum quis nulla quis, convallis
          semper mi.
        </p>
      </div>
      <div slot="webcl-card-footer">
        <p>WebCl Card Footer</p>
      </div>
    </ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="using all the defined slots" args={cardWithCustomWidthHeight}>
    {withAllDefinedSlots.bind({})}
  </Story>
</Canvas>

Basic card using custom slot name

export const cardWithCustomSlotNameTemplate = ({
  hasRoundedBorder,
  cardWidth,
  cardHeight,
  customSlotName,
  showCustomSlot,
}) =>
  html`<div class="storybook-card-container">
    <ymlwebcl-card
      .hasRoundedBorder=${hasRoundedBorder}
      .cardHeight=${cardHeight}
      .cardWidth=${cardWidth}
      .showCustomSlot=${showCustomSlot}
      .customSlotName=${customSlotName}
    >
      <div slot="custom-card-slot">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
          ac aliquam orci. In dignissim eget enim ut faucibus. Donec volutpat
          libero a ipsum tincidunt, in vehicula tellus congue. Mauris id
          consequat justo. Donec turpis justo, dictum quis nulla quis, convallis
          semper mi.
        </p>
      </div>
    </ymlwebcl-card>
  </div>`;

<Canvas>
  <Story name="card with custom slot name" args={cardWithCustomSlot}>
    {cardWithCustomSlotNameTemplate.bind({})}
  </Story>
</Canvas>
