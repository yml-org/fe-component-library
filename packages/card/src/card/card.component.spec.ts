import { CardComponent } from './card.component';

describe('ymlwebcl-card', () => {
  const CARD_COMPONENT = 'ymlwebcl-card';
  let cardElement: CardComponent;

  const getShadowRoot = (tagName: string): ShadowRoot | null =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    cardElement = window.document.createElement(
      CARD_COMPONENT
    ) as CardComponent;
    document.body.appendChild(cardElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(CARD_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(CARD_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('renders the card with rounded border', async () => {
    cardElement['hasRoundedBorder'] = true;
    await cardElement.updateComplete;
    const cardComponentClassList =
      getShadowRoot(CARD_COMPONENT)?.querySelector('div')?.classList;
    expect(cardComponentClassList).toContain('rounded-lg');
  });

  it('renders the card with custom width', async () => {
    const CUSTOM_WIDTH = 200;
    cardElement['cardWidth'] = CUSTOM_WIDTH;
    await cardElement.updateComplete;
    const cardComponentStyle =
      getShadowRoot(CARD_COMPONENT)?.querySelector('div')?.style;
    expect(cardComponentStyle?.width).toBe(`${CUSTOM_WIDTH}px`);
  });

  it('renders the card with custom height', async () => {
    const CUSTOM_HEIGHT = 200;
    cardElement['cardHeight'] = CUSTOM_HEIGHT;
    await cardElement.updateComplete;
    const cardComponentStyle =
      getShadowRoot(CARD_COMPONENT)?.querySelector('div')?.style;
    expect(cardComponentStyle?.minHeight).toBe(`${CUSTOM_HEIGHT}px`);
  });

  it('renders the component passed in slot', async () => {
    cardElement['customSlotName'] = 'card-slot';
    cardElement['showCustomSlot'] = true;
    window.document.body.appendChild(cardElement);
    (document.querySelector('ymlwebcl-card') || document.body).innerHTML =
      '<div slot="card-slot" class="slot-test">Test</div>';
    await cardElement.updateComplete;
    const elemClassList = document
      ?.querySelector('ymlwebcl-card')
      ?.querySelector('div')?.classList;
    expect(elemClassList).toContain('slot-test');
  });

  it('renders content in row if isHorizontal flag is set to true', async () => {
    cardElement['isHorizontal'] = true;
    await cardElement.updateComplete;
    const cardComponentClassList =
      getShadowRoot(CARD_COMPONENT)?.querySelector('div')?.classList;
    expect(cardComponentClassList).toContain('flex-row');
  });
});
