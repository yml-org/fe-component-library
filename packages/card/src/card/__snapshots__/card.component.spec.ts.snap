// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-component matches component snapshot 1`] = `
v {
  "_$AL": Map {},
  "_$Do": N {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": gt {
      "_$AD": H {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "style",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 1,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        U {
          "_$AH": "webcl-card-wrapper flex flex-col",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-card-wrapper flex flex-col"
            part="webcl-card-container"
            style="width:300px; min-height:300px"
          >
            
      
            <!--?lit$472687944$-->
            <div
              class="webcl-card-header-section"
              part="webcl-card-header"
            >
              
            
              <slot
                name="webcl-card-header"
              >
                
          
              </slot>
            </div>
            
          
            <div
              class="webcl-card-content-section"
              part="webcl-card-content"
            >
              
            
              <slot
                name="webcl-card-content"
              >
                
          
              </slot>
            </div>
            
          
            <div
              class="webcl-card-footer-section mt-auto"
              part="webcl-card-footer"
            >
              
            
              <slot
                name="webcl-card-footer"
              >
                
          
              </slot>
            </div>
            
    
          </div>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        U {
          "_$AH": "width:300px; min-height:300px",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="webcl-card-wrapper flex flex-col"
            part="webcl-card-container"
            style="width:300px; min-height:300px"
          >
            
      
            <!--?lit$472687944$-->
            <div
              class="webcl-card-header-section"
              part="webcl-card-header"
            >
              
            
              <slot
                name="webcl-card-header"
              >
                
          
              </slot>
            </div>
            
          
            <div
              class="webcl-card-content-section"
              part="webcl-card-content"
            >
              
            
              <slot
                name="webcl-card-content"
              >
                
          
              </slot>
            </div>
            
          
            <div
              class="webcl-card-footer-section mt-auto"
              part="webcl-card-footer"
            >
              
            
              <slot
                name="webcl-card-footer"
              >
                
          
              </slot>
            </div>
            
    
          </div>,
          "name": "style",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        N {
          "_$AA": <!--?lit$472687944$-->,
          "_$AB": 
    ,
          "_$AH": gt {
            "_$AD": H {
              "el": <template />,
              "parts": Array [],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "u": Array [],
          },
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__cardDimensionUnit": "px",
  "__cardHeight": 300,
  "__cardWidth": 300,
  "__customSlotName": "",
  "__hasRoundedBorder": false,
  "__isHorizontal": false,
  "__showCustomSlot": false,
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": N {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": gt {
        "_$AD": H {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "style",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 1,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          U {
            "_$AH": "webcl-card-wrapper flex flex-col",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-card-wrapper flex flex-col"
              part="webcl-card-container"
              style="width:300px; min-height:300px"
            >
              
      
              <!--?lit$472687944$-->
              <div
                class="webcl-card-header-section"
                part="webcl-card-header"
              >
                
            
                <slot
                  name="webcl-card-header"
                >
                  
          
                </slot>
              </div>
              
          
              <div
                class="webcl-card-content-section"
                part="webcl-card-content"
              >
                
            
                <slot
                  name="webcl-card-content"
                >
                  
          
                </slot>
              </div>
              
          
              <div
                class="webcl-card-footer-section mt-auto"
                part="webcl-card-footer"
              >
                
            
                <slot
                  name="webcl-card-footer"
                >
                  
          
                </slot>
              </div>
              
    
            </div>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          U {
            "_$AH": "width:300px; min-height:300px",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="webcl-card-wrapper flex flex-col"
              part="webcl-card-container"
              style="width:300px; min-height:300px"
            >
              
      
              <!--?lit$472687944$-->
              <div
                class="webcl-card-header-section"
                part="webcl-card-header"
              >
                
            
                <slot
                  name="webcl-card-header"
                >
                  
          
                </slot>
              </div>
              
          
              <div
                class="webcl-card-content-section"
                part="webcl-card-content"
              >
                
            
                <slot
                  name="webcl-card-content"
                >
                  
          
                </slot>
              </div>
              
          
              <div
                class="webcl-card-footer-section mt-auto"
                part="webcl-card-footer"
              >
                
            
                <slot
                  name="webcl-card-footer"
                >
                  
          
                </slot>
              </div>
              
    
            </div>,
            "name": "style",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          N {
            "_$AA": <!--?lit$472687944$-->,
            "_$AB": 
    ,
            "_$AH": gt {
              "_$AD": H {
                "el": <template />,
                "parts": Array [],
              },
              "_$AM": [Circular],
              "_$AN": undefined,
              "u": Array [],
            },
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
